File: sieve.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

bool is_prime(long long n)
{
    for (long long i = 2; i < n; i++)
    {
        if (n % i == 0)
        {
            return false;
        }
    }
    return true;
}

bool sieve(long long n)
{
    long long i = 2;
    while (i * i <= n)
    {
        if (n % i == 0)
        {
            return false; // the number is NOT prime
        }

        do
        {
            i++;
        } while (!is_prime(i));
    }
    return true; // the number is prime
}

int main()
{
    int test_cases;
    long long k;
    scanf("%lld", &test_cases);

    while (test_cases--)
    {
        scanf("%lld", &k);
        printf("%d %s\n", k, sieve(k) ? "True" : "False");
    }

    return 0;
}

File: sieve.py
def is_prime(n):
    i = 2
    while i < n:
        if n % i == 0:
            return False
        i += 1
    return True

def next_prime(n):
    while True:
        n += 1
        if is_prime(n):
            break
    return n

def sieve(k):
    i = 2
    while i * i <= k:
        if k % i == 0:
            return False # the number is NOT prime
        i = next_prime(i)   
    return True # the number is prime


if __name__ == "__main__":
    test_cases = int(input())
    for i in range(test_cases):
        k = int(input())
        print(k, sieve(k))
